src/Commands/TideSearchCommands.php

```diff
@@ -0,0 +1,160 @@
+      $ops = "Not in index";
+      $description = "published but not indexed in the search";
+      if (!empty($not_in_index)) {
+        $not_in_index = implode(", ", $not_in_index);
+        self::auditIntoLog($not_in_index, $ops, $description);
+        $message = $message . "The following nodes are published but not indexed in the search - " . $not_in_index . "\n";
+      }
+      // Content that is in the search index but is NOT published.
+      $not_published = array_diff($nids_in_search_index, $nids_of_published_content);
+      $ops = "Not published";
+      $description = "in search index but not published";
+      if (!empty($not_published)) {
+        $not_published = implode(", ", $not_published);
+        self::auditIntoLog($not_published, $ops, $description);
+        $message = $message . "The following nodes are in search index but not published - " . $not_published . "\n";
+      }
+      $message = ($not_published || $not_in_index) ? $message . "Logged in audit trail as well." : "Nothing to log.";
+      return $message;
+    }
+    catch (ConsoleException $exception) {
+      throw new \Exception($exception->getMessage());
+    }
+  }
+
+  /**
+   * Helper to log into the audit trail.
+   */
+  public static function auditIntoLog($nids, $ops, $description) {
+    $log = [
+      "type" => "tide_search",
+      "operation" => $ops,
+      "description" => t("The following nodes are %des - %nids", [
+        "%nids" => $nids,
+        "%des" => $description,
+      ]),
+      "ref_numeric" => 1,
+      "ref_char" => "drush sapi-nsc node results",
+    ];
+    // Add the log to the "admin_audit_trail" table.
+    if (function_exists("admin_audit_trail_insert")) {
+      admin_audit_trail_insert($log);
+    }
+  }
+
+  /**
+   * Helper to get node Ids from search index.
+   */
+  public static function getNidsFromSearchIndex($indexId) {
+    $command_helper = new CommandHelper(\Drupal::entityTypeManager(), \Drupal::moduleHandler(), \Drupal::service('event_dispatcher'), 'dt');
+    $es_nids = [];
+    $indexes = $command_helper->loadIndexes([$indexId]);
+    if (empty($indexes[$indexId])) {
+      throw new ConsoleException(t('@index was not found'));
+    }
+    $total = $indexes[$indexId]->getTrackerInstance()->getTotalItemsCount();
+    $no_of_batches = ceil($total / 10000);
+    // If the result set is more than 10000 then run it in batch.
+    if ($no_of_batches > 1) {
+      $nid_starting_point = 0;
+      for ($i = 1; $i <= $no_of_batches; $i++) {
+        try {
+          $query = $indexes[$indexId]->query();
+          $query->range(0, 10000);
+          $query->sort('nid');
+          $query->addCondition('nid', $nid_starting_point, '>');
+          $results = $query->execute();
+        }
+        catch (SearchApiException $e) {
+          throw new ConsoleException($e->getMessage(), 0, $e);
+        }
+        foreach ($results->getResultItems() as $item) {
```

Comment:
hey @MdNadimHossain ,
I think you don't need the function `convertToNodeIds`.
the `$results` is an object that implements `ResultInterface`, so I think you can get the actual result by using `$results->getExtraData('elasticsearch_response')' or `$results->getAllExtraData()`, then loop through to get `nid` from the result.